symengine/subs.h

```diff
@@ -192,7 +192,39 @@ class XReplaceVisitor : public BaseVisitor<XReplaceVisitor>
                 throw SymEngineException("expected an object of type Boolean");
             v.insert(rcp_static_cast<const Boolean>(a));
         }
-        result_ = x.create(v);
+        result_ = logical_and(v);
+    }
+
+    void bvisit(const Or &x)
+    {
+        set_boolean v;
+        for (const auto &elem : x.get_container()) {
+            auto a = apply(elem);
+            if (not is_a_Boolean(*a))
```

Comment:
Is it possible to construct an `Or` where `a` is not a `Boolean`?
(I see this check was also in `And`, but I don't see how to write a test case that triggers this exception)

Reply:
I can't see how you would either - would think it would only be the result of a framework bug that allowed you to create a faulty Boolean instance. I guess the check arguably gives you a slightly better exception than the static_cast would but if you think better to just remove them all happy to.

Reply:
Perhaps since `a` was created after the `apply` something bad could have been created by the apply given bad input to subs.  What would happen say if we have `logical_and(x, y)` and we call  `subs`  with `x=2` and `y=2` for example. Would this exception be triggered in this case?

Reply:
I don't think you *can* have a `logical_and(x,y)` currently - the only constructor I see for `And` takes a `set_boolean`.
But I guess it makes sense to keep the check, in case this becomes possible in the future. (especially since I think the static cast could potentially segfault rather than raise an exception)